lms/djangoapps/shoppingcart/tests/test_reports.py

```diff
@@ -229,7 +230,7 @@ def test_purchased_csv(self):
         report = initialize_report("itemized_purchase_report", self.now - self.FIVE_MINS, self.now + self.FIVE_MINS)
         csv_file = StringIO()
         report.write_csv(csv_file)
-        csv = csv_file.getvalue()
+        csv = csv_file.getvalue() if six.PY3 else csv_file.getvalue().decode('utf-8') 
```

Comment:
Can we update the upstream function to always produce a csv object that always provides unicode values?

Reply:
Still waiting for an answer on this.

Reply:
did not updated it yet

Reply:
@Ayub-Khan what I meant was updating the update report generation tool to initialize a CSV from a  unicode string?

Reply:
@feanil the report generation tool generates unicode. But we are using the six.StringIO which is a bytes buffer in py2. when we write using unicodecsv it writes it as bytes and when reading it reads as default python2 bytes string. while in python3 it uses stringio which takes unicode string inputs and writes them as unicode input and reads as unicode strings. I think we are just writing and reading in memory buffer for testing. In really when a real file object is given it will be written as unicode as our out put is already unicode.

Reply:
Ah, that makes a lot of sense, thank you!  Please put the above explaination as a comment above this line in the code.